std/algorithm/iteration.d

```diff
@@ -5034,28 +5039,28 @@ if (isSomeChar!C)
         }
 
     public:
-        this(C[] s) pure @safe
+        this(Range s)
         {
-            import std.string : strip;
-            _s = s.strip();
+            import std.string : stripLeft;
+            _s = s.stripLeft();
             getFirst();
         }
 
-        @property C[] front() pure @safe
+        @property front()
```

Comment:
Why did you remove the return type here?

Reply:
It's `auto` return now. I could put `Range`, but that may be more confusing.

Plus, I know this is a template type, but not sure if attributes will be inferred if it's not `auto`, and I can't specify `pure @safe` anymore, due to the possibility of the underlying range not being one of those.

Reply:
In any case, the type is voldemort, it has no docs except in the doc of the function. So including a type has no effect on documentation. One has to look at the implementation to see what it is returning exactly, and the answer is right below anyway.

Reply:
More info -- this must be inferred, but I was wrong that it's not inferred without an `auto` return. Otherwise, the other functions wouldn't be inferred, and the `pure @safe` unittest I added below would fail to compile.

So I can add the return type back in, if that's what gets this merged.

Reply:
I was just curious because I think dscanner has a rule (not enabled here) that enforces that the return type is either defined or auto.

Reply:
Done.